code/game/machinery/dance_machine.dm

```diff
@@ -127,6 +127,9 @@
 				to_chat(usr, "<span class='warning'>Error: You cannot change the song until the current one is over.</span>")
 				return
 			check_GBP()
+			var/list/available = list()
+			for(var/datum/track/S in songs)
+				available += S.name
```

Comment:
@Robustin When you add point requirements do a if(!(points >= S.gbp_required)) continue or something, because this will work better as it's not hardcoded.